pwsh_module/pwsh_bolt_internal.ps1

```diff
@@ -42,7 +42,19 @@ function Get-BoltCommandline {
       Write-Verbose "Parsing $($kvp.key) as hashtable parameter"
       $v = ConvertTo-Json -InputObject $pwshValue -Compress
       $params += "--$($rubyParameter)"
-      $params += "'$($v)'"
+      # Commands run on Linux platforms will fail because the
+      # the interpolation of the command from powershell -> .NET -> Ruby
+      # will end up stripping out quotes that aren't baskslash-escaped.
+      #
+      # See https://github.com/MicrosoftDocs/PowerShell-Docs/issues/2361
+      #
+      # $IsLinux is an automatic variable from powershellcore:
+      # https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-7.1#islinux
+      if ($IsLinux) {
+        $params += $v -replace '"', '\"'
+      } else {
+        $params += "'$($v)'"
+      }
```

Comment:
```suggestion
     $IsLinuxEnv = (Get-Variable -Name "IsLinux" -ErrorAction Ignore) -and $IsLinux
     $IsMacOSEnv = (Get-Variable -Name "IsMacOS" -ErrorAction Ignore) -and $IsMacOS
     $IsWinEnv = !$IsLinuxEnv -and !$IsMacOSEnv
     if($IsWinEnv){
       $params += "'$($v)'"
     } else {
       $params += $v -replace '"', '\"'
     }
```

Reply:
I think we'll need the get-variable pattern to handle being run in 5.1, and we'll need to account for macox not being detected as linux.

Reply:
Updated